pkg/helpers/clustersetmapper.go

```diff
@@ -0,0 +1,96 @@
+package helpers
+
+import (
+	"sync"
+)
+
+type ClusterSetMapper struct {
+	mutex sync.RWMutex
+	// mapping: ClusterSet - Clusters
+	clusterSet2Clusters map[string][]string
+	// mapping: Cluster - ClusterSets
+	cluster2ClusterSets map[string][]string
+}
+
+func NewClusterSetMapper() *ClusterSetMapper {
+	return &ClusterSetMapper{
+		clusterSet2Clusters: make(map[string][]string),
+		cluster2ClusterSets: make(map[string][]string),
+	}
+}
+
+func (c *ClusterSetMapper) UpdateClusterSetByClusters(clusterSetName string, clusters []string) {
+	if clusterSetName == "" {
+		return
+	}
+
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	for _, oldCluster := range c.clusterSet2Clusters[clusterSetName] {
+		c.deleteClusterSetOfCluster(oldCluster, clusterSetName)
+	}
+
+	c.clusterSet2Clusters[clusterSetName] = clusters
+	for _, cluster := range clusters {
+		c.cluster2ClusterSets[cluster] = append(c.cluster2ClusterSets[cluster], clusterSetName)
+	}
+	return
+}
+
+func (c *ClusterSetMapper) DeleteClusterSet(clusterSetName string) {
+	if clusterSetName == "" {
+		return
+	}
+
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	for _, oldCluster := range c.clusterSet2Clusters[clusterSetName] {
+		c.deleteClusterSetOfCluster(oldCluster, clusterSetName)
+	}
+	delete(c.clusterSet2Clusters, clusterSetName)
+
+	return
+}
+
+func (c *ClusterSetMapper) GetClustersOfClusterSet(clusterSetName string) []string {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	return c.clusterSet2Clusters[clusterSetName]
+}
+
+func (c *ClusterSetMapper) GetAllClusterSet2Clusters() map[string][]string {
```

Comment:
s/GetAllClusterSet2Clusters/GetAllClusterSetToClusters/g

maybe do not use  numbers here.